activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -65,7 +65,7 @@ def eagerly_generate_alias_attribute_methods(_new_name, _old_name) # :nodoc:
       end
 
       def generate_alias_attributes # :nodoc:
-        superclass.generate_alias_attributes unless base_class?
+        superclass.generate_alias_attributes unless superclass == Base
```

Comment:
cc: @ipc103 

Sorry I was the one suggesting to use `base_class?` in https://github.com/rails/rails/pull/48913#discussion_r1295905312 but I had wrong understanding of its behavior so `superclass` compared to `AR::Base` is the more correct check here, as you initially did!

Reply:
I definitely missed that as well! Thank you @andrewn617 for the quick fix ❤️